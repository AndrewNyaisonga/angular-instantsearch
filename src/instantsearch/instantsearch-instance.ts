import { Injectable } from "@angular/core";
import instantsearch from "instantsearch.js/es";

@Injectable()
export class NgISInstance {
  public appId: string;
  public apiKey: string;
  public indexName: string;

  public numberLocale?: string;
  public searchFunction?: () => void;
  public createAlgoliaClient?: () => object;
  public searchParameters?: object | void;
  public urlSync?:
    | boolean
    | {
        mapping?: object;
        threshold?: number;
        trackedParameters?: string[];
        useHash?: boolean;
        getHistoryState?: () => object;
      };

  private instance?: {
    start: () => void;
  };

  public init(config: object) {
    this.instance = instantsearch(config);
  }

  public start() {
    this.instance.start();
  }
}
