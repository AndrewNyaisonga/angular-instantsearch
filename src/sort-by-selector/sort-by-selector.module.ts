import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";

import { NgISFooterModule } from "../footer/footer.module";
import { NgISHeaderModule } from "../header/header.module";
import { NgISSortBySelector } from "./sort-by-selector";

@NgModule({
  declarations: [NgISSortBySelector],
  entryComponents: [NgISSortBySelector],
  exports: [NgISSortBySelector],
  imports: [CommonModule, NgISHeaderModule, NgISFooterModule]
})
export class NgISSortBySelectorModule {}
