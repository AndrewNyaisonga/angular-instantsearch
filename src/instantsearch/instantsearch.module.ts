import { CommonModule } from "@angular/common";
import { ModuleWithProviders, NgModule } from "@angular/core";

import { NgISInstantSearch } from "./instantsearch";
import { NgISInstance } from "./instantsearch-instance";

@NgModule({
  declarations: [NgISInstantSearch],
  entryComponents: [NgISInstantSearch],
  exports: [NgISInstantSearch],
  imports: [CommonModule]
})
export class NgISInstantSearchModule {
  public static forRoot(): ModuleWithProviders {
    return {
      ngModule: NgISInstantSearchModule,
      providers: [NgISInstance]
    };
  }
}
