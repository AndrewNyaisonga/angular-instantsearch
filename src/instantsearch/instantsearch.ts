import { AfterViewInit, Component, Input, OnInit } from "@angular/core";

import { NgISInstance } from "./instantsearch-instance";

@Component({
  selector: "ngis-instantsearch",
  template: `<ng-content></ng-content>`
})
export class NgISInstantSearch implements AfterViewInit, OnInit {
  @Input() public config: InstantSearchConfig;

  constructor(private searchInstance: NgISInstance) {}

  public ngOnInit() {
    this.searchInstance.init(this.config);
  }

  public ngAfterViewInit() {
    this.searchInstance.start();
  }
}
