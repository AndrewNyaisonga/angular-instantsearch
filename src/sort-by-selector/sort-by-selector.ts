import { Component, Input } from "@angular/core";
import { connectSortBySelector } from "instantsearch.js/es/connectors";
import { noop } from "lodash-es";

import { BaseWidget } from "../base-widget";
import { NgAisInstance } from "../instantsearch/instantsearch-instance";

interface State {
  currentRefinement?: string;
  options: {}[];
  refine: Function;
}

@Component({
  selector: "ng-ais-sort-by-selector",
  template: `
    <div [class]="cx()">
      <ng-ais-header [header]="header" [className]="cx('header')"></ng-ais-header>

      <div [class]="cx('body')">
        <select
          [class]="cx('select')"
          (change)="state.refine($event.target.value)"
        >
          <option
            [class]="cx('option')"
            *ngFor="let item of state.options"
            [value]="item.value"
            [selected]="item.value === state.currentRefinement"
          >
            {{item.label}}
          </option>
        </select>
      </div>

      <ng-ais-footer [footer]="footer" [className]="cx('footer')"></ng-ais-footer>
    </div>
  `
})
export class NgAisSortBySelector extends BaseWidget {
  @Input()
  public indices: {
    name: string;
    label: string;
  }[];

  public state: State = {
    currentRefinement: null,
    options: [],
    refine: noop
  };

  constructor(searchInstance: NgAisInstance) {
    super(searchInstance, "SortBySelector");
  }

  public ngOnInit() {
    this.createWidget(connectSortBySelector, { indices: this.indices });
    super.ngOnInit();
  }
}
