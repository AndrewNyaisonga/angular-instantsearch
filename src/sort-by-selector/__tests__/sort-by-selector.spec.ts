import { TestBed } from "@angular/core/testing";

import { NgAisInstantSearchModule } from "../../instantsearch/instantsearch.module";
import { NgAisSortBySelectorModule } from "../sort-by-selector.module";
import { NgAisSortBySelector } from "../sort-by-selector";

jest.mock("../../base-widget");

const defaultState = {
  currentRefinement: "foo",
  options: [{ label: "foo", value: "foo" }, { label: "bar", value: "bar" }],
  refine: jest.fn()
};

const render = (state?: {}) => {
  const fixture = TestBed.createComponent(NgAisSortBySelector);

  if (state) {
    fixture.componentInstance.updateState({ ...defaultState, ...state }, false);
  }

  fixture.detectChanges();
  return fixture;
};

describe("SortBySelector", () => {
  beforeEach(() =>
    TestBed.configureTestingModule({
      declarations: [],
      imports: [NgAisInstantSearchModule.forRoot(), NgAisSortBySelectorModule]
    })
  );

  it("renders markup without state", () => {
    const fixture = render();
    expect(fixture).toMatchSnapshot();
  });

  it("renders markup with state", () => {
    const fixture = render({});
    expect(fixture).toMatchSnapshot();
  });

  it("should call refine on select change", () => {
    const refine = jest.fn();
    const fixture = render({ refine });

    const select = fixture.debugElement.nativeElement.querySelector("select");
    select.value = "bar";
    select.dispatchEvent(new Event("change"));

    expect(refine).toHaveBeenCalled();
    expect(refine).toHaveBeenCalledWith("bar");
  });
});
