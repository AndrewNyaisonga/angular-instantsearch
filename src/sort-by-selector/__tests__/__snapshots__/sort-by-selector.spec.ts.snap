// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBySelector renders markup with state 1`] = `
<ng-ais-sort-by-selector
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-SortBySelector"
  >
    
        
    <ng-ais-header
      classname="ais-SortBySelector-header ais-header"
      ng-reflect-class-name="ais-SortBySelector-header ais-"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-SortBySelector-body ais-body"
    >
      
          
      <select
        class="ais-SortBySelector-select"
      >
        
            
        
        <option
          class="ais-SortBySelector-option"
          value="foo"
        >
          
              foo
            
        </option>
        <option
          class="ais-SortBySelector-option"
          value="bar"
        >
          
              bar
            
        </option>
        
          
      </select>
      
        
    </div>
    
  
        
    <ng-ais-footer
      classname="ais-SortBySelector-footer ais-footer"
      ng-reflect-class-name="ais-SortBySelector-footer ais-"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-sort-by-selector>
`;

exports[`SortBySelector renders markup without state 1`] = `
<ng-ais-sort-by-selector
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-SortBySelector"
  >
    
        
    <ng-ais-header
      classname="ais-SortBySelector-header ais-header"
      ng-reflect-class-name="ais-SortBySelector-header ais-"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-SortBySelector-body ais-body"
    >
      
          
      <select
        class="ais-SortBySelector-select"
      >
        
            
        
        
          
      </select>
      
        
    </div>
    
  
        
    <ng-ais-footer
      classname="ais-SortBySelector-footer ais-footer"
      ng-reflect-class-name="ais-SortBySelector-footer ais-"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-sort-by-selector>
`;
